packages/cli/src/lib/snapshot/snapshot.ts

```diff
@@ -8,7 +15,19 @@ export class Snapshot {
   ) {}
 
   async validate() {
-    // TODO: CDX-358: Validate snapshot
+    await this.client.dryRun(this.snapshotId, {
+      deleteMissingResources: false, // TODO: CDX-361: Add flag to support missing resources deletion
+    });
+
+    await this.waitUntilIdle();
+  }
+
+  isValid() {
```

Comment:
The class interface here is a bit weird, because you have to call `validate()`, then you can call `isValid()` afterwards.

As a user of this class, you could very well try to call `isValid()` without first doing `validate()`, and then you'd get incorrect results.

I think we could refactor this so `validate()` returns true or false, and we could remove the `isValid()` method.